dateutil/rrule.py

```diff
@@ -1664,6 +1664,8 @@ def _parse_rfc(self, s,
                     name = "RRULE"
                     value = line
                 else:
+                    if line.count(':') > 1:
+                        raise ValueError("Malformed rule: " + line)
```

Comment:
While checking for the number of colons works, I am not sure it is the best approach. If I only use a colon for the timezone offset, we get the confusing error again:

```pycon
>>> rrulestr("FREQ=YEARLY;UNTIL=20190329T005959+01:00",dtstart=start)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/Daniel/ffe4/dateutil/dateutil/rrule.py", line 1734, in __call__
    return self._parse_rfc(s, **kwargs)
  File "/Users/Daniel/ffe4/dateutil/dateutil/rrule.py", line 1702, in _parse_rfc
    raise ValueError("unsupported property: " + name)
ValueError: unsupported property: FREQ=YEARLY
```
After looking at the code again, I think making this a separate error message might be a little more work than I thought at first. I think it would be okay to go back to the previous version, replacing the catch-all error message, but simplify the message a bit. Rather than referring to the RFC and some unspecified delimiter, we can just say that there cannot be any colon other than the one separating the property name. I also think it makes more sense to return the specific line, as you did here, rather than the complete rule.

Reply:
I understand your point. One thought I have: In addition to checking count of ':', can I also check the pattern for name before checking if it is equal to "RRULE" like this:
```
valid_name_pattern = re.compile("^[A-Z]+$")
if not valid_name_pattern.match(name):
      raise ValueError("Malformed rule: " + line)
elif name == "RRULE":
```

Reply:
I think you could do something similar. When we arrive at the catch-all error, we can add a check for whether we have a valid property name. If it is a valid property name we return that it is not supported, otherwise we return that we could not correctly parse the line (likely because of colons).

The pattern for the property name should also contain the hyphen `-` and maybe lower case letters for good measure (see [section 3.1](https://tools.ietf.org/html/rfc5545#section-3.1) and section [3.8.8.2](https://tools.ietf.org/html/rfc5545#section-3.8.8.2) for valid property names)

Of course, we could also check for a colon-separated datetime via the same method and return a helpful error message for that as well. But I think it is enough do mention datetimes in the error message for invalid property names.

Reply:
I have checked for specific cases as per your comment and made a commit. Please review. Thank you